src/Bicep.LangServer/Snippets/Templates/res-loadbalancer-internal.bicep

```diff
@@ -1,52 +1,65 @@
 ﻿// Internal Load Balancer
-resource /*${1:loadBalancerInternal}*/loadBalancerInternal 'Microsoft.Network/loadBalancers@2020-11-01' = {
-  name: /*${2:'name'}*/'name'
+resource vnet 'Microsoft.Network/virtualNetworks@2021-02-01' existing = {
+  name: /*${1:'name'}*/'name'
+}
+
+resource /*${2:'subnet'}*/subnet 'Microsoft.Network/virtualNetworks/subnets@2021-02-01' existing = {
+  parent: vnet
+  name: /*${3:'name'}*/'name'
+}
+
+resource /*${4:backendAddressPool}*/backendAddressPool 'Microsoft.Network/loadBalancers/backendAddressPools@2021-02-01' existing = {
+  name: /*${5:'loadBalancerExternal/loadBalancerBackEndPool'}*/'loadBalancerExternal/loadBalancerBackEndPool'
+}

```

Comment:
I don't think it makes sense to include this in the snippet itself - most of the time I'd expect the user to already have references to these resources.

I'd expect the snippet to be either:
```bicep
          subnet: {
            id: /*${5:subnetRef}*/subnetRef.id
          }
```

Or:
```bicep
          subnet: {
            id: /*${5:subnetId}*/subnetId
          }
```

Reply:
If we don't include the snippet, then we'll have to do below(include single quotes) to avoid errors:
```
subnet: {
   id: /*${5:'subnetId'}*/'subnetId'
}
```
We expect user to remove quotes and update the reference. Is that correct?

Reply:
I don't understand why we'd want to keep the quotes in the snippet - does it matter whether or not the snippet itself is error-free? I feel like it would be better to cater to the snippet user rather than the snippet author, and I can't think of any scenarios where quotes would be desirable for an `id` field.

Reply:
Hmm my understanding was we wanted to keep the snippet error-free.

As part of https://github.com/Azure/bicep/issues/2944, I even added test to enforce this
https://github.com/Azure/bicep/blob/f650dc850f65742c980590ebe3835fa197a85bc3/src/Bicep.LangServer.IntegrationTests/SnippetTemplatesTests.cs#L27

Correct me if I am wrong.

Adding @MarcusFelling and @majastrz as well.

Reply:
As discussed offline, removed referenced snippets:
https://github.com/Azure/bicep/commit/c4c04eb3bc9a0d489686c47d104a3ee22178b142

Also, included quotes as part of snippet placeholder:
![image](https://user-images.githubusercontent.com/30270536/127570425-21e15b41-3f7f-4c88-ba02-ab85701c7d80.png)